ui/components/DashboardComponent.js

```diff
@@ -195,6 +225,64 @@ class DashboardComponent extends React.Component {
     );
   };
 
+  fetchMeshScanData = () => {
+    const self = this;
+    self.props.updateProgress({ showProgress: true });
+    dataFetch(
+      "/api/mesh/scan",
+      {
+        credentials: "same-origin",
+        method: "GET",
+        credentials: "include",
+      },
+      (result) => {
+        self.props.updateProgress({ showProgress: false });
+        if (result) {
+          // Extract all the unique namespaces in the mesh scan
+          const namespaces = new Set();
+          Object.keys(result).forEach(mesh => {
+            if (Array.isArray(result[mesh])) {
+              result[mesh].forEach(comp => {
+                comp.metadata && namespaces.add(comp.metadata.namespace)
+              })
+            }
+          })
+          self.setState({ 
+            meshScanNamespaces: [...namespaces], 
+            activeMeshScanNamespace: [...namespaces][0] || "" 
+          });
+
+          // Check if Istio data is present in the scan
+          if (Array.isArray(result.Istio)) {
+            const istioData = result.Istio.map(comp => {
+              const compData = {}
+              if(comp.metadata?.name === "istio-egressgateway") {
```

Comment:
Why the hardcoding? By this you are limiting the data being display, query over the array, and display metadata.name for each service detected.

Reply:
Actually this was kind of hardcoded because I couldn't figure out a way to convert `istio-egressgateway` to `Istio Egress` as it was in the mockup and same goes for ingress gateway. Should I add a default case where the `comp.metadata.name` is displayed when moment of the cases match?

Reply:
See the above review